spring-kafka/src/main/java/org/springframework/kafka/listener/ErrorHandlingUtils.java

```diff
@@ -120,7 +122,13 @@ public static void retryBatch(Exception thrownException, ConsumerRecords<?, ?> r
 			Exception lastException = unwrapIfNeeded(thrownException);
 			Boolean retryable = classifier.classify(lastException);
 			while (Boolean.TRUE.equals(retryable) && nextBackOff != BackOffExecution.STOP) {
-				consumer.poll(Duration.ZERO);
+				try {
+					consumer.poll(Duration.ZERO);
+				}
+				catch (WakeupException we) {
+					seeker.handleBatch(thrownException, records, consumer, container, () -> { });
+					throw new KafkaException("Woken up during retry", logLevel, we);
```

Comment:
I don't think we need to re-seek and throw an exception here, we can just ignore the wakeup (perhaps try another poll to keep the broker from triggering a rebalance because this poll failed.

Reply:
On the other hand, since this is such a rare race condition, I guess doing a seek under these circumstances will not significantly affect performance.

Reply:
Yes, I've just tried it and the test is passing, but shouldn't we bail-out from retrying because my idea of pausng container is to suspend any new traffic and also suspend re-attempts.

Reply:
Good point; ok; will merge and back-port tomorrow.

Reply:
Now when I think more about what I just wrote, I realized that this retry loop doesn't check about container's _pause_ state at all. It does check `isRunning()` but not `isPaused()/isPauseRequsted()`, and retries will continue.

 Is that how it's supposed to behave or it should be addressed as separate issue? (seems like it's a bit out-of-scope of this PR)

Reply:
Right - I just had the same thought and was going to comment here - we need to add some kind of hook to exit early if the container is paused; checking the paused state should work. 

But I agree, it probably should be a different issue/PR.

We might want to consider it also for when individual partitions are paused (and the retrying batch contains records from those partitions) - but maybe that's a stretch too far.

Reply:
@antonio-tomac I created an issue for this: https://github.com/spring-projects/spring-kafka/issues/2742 if you want to take a stab at it, and it is relatively simple, we might be able to squeeze it into Monday's releases.